tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -1103,12 +1110,14 @@ void TangoRosNode::PublishMesh() {
             &t3dr_updated_indices);
       }
       // Publish Tango mesh as visualization marker.
-      if (mesh_marker_publisher_.getNumSubscribers() > 0) {
+      if (enable_3dr_mesh_) {
         visualization_msgs::MarkerArray mesh_marker_array;
         tango_3d_reconstruction_helper::ExtractMeshAndConvertToMarkerArray(
             t3dr_context_, t3dr_updated_indices, time_offset_,
             start_of_service_frame_id_, &mesh_marker_array);
-        mesh_marker_publisher_.publish(mesh_marker_array);
+        if (mesh_marker_publisher_.getNumSubscribers() > 0) {
```

Comment:
Can we move this check around the Extract function? No need to extract anything as long as we are not publishing?